### ISSUE TITLE
RDS provider should allow optional username and password when snapshot_identifier is provided

### ISSUE THREAD

**Issue by leetrout** [hashicorp/terraform#3368]
Based on https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go#L306 when `snapshot_identifier` is provided the RestoreDBInstanceFromDBSnapshotInput endpoint is used and username and password should not be required.

**Comment by stack72** [comment]
@catsby I suggest we change the schema to reflect this as follows:

```hcl
      "username": &schema.Schema{
				Type:          schema.TypeString,
				Optional:      true,
				ForceNew:      true,
        ConflictsWith: []string{"snapshot_identifier"},
			},
			"password": &schema.Schema{
				Type:     schema.TypeString,
				Required: true,
			},
      "snapshot_identifier": &schema.Schema{
				Type:     schema.TypeString,
				Computed: false,
				Optional: true,
				Elem:     &schema.Schema{Type: schema.TypeString},
			},
```

This way we can validate for either username or snapshot_identifier. We don't need both. Use of snapshot identifier has a very specific use-case

Thoughts?

**Comment by stack72** [comment]
Hi @leetrout 

My apologies for this being missed, but this issue was resolved in 45ef11e59f443c43b42fe2d36c639c55bb0b1fbd

Hope this helps

Paul

**Comment by leetrout** [comment]
Awesome! thanks @stack72
